modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/type/Subscription.kt

```diff
@@ -17,22 +17,22 @@ sealed interface Subscription {
     fun numFreeThenPricePerPeriod(res: Resources): String?
     fun tryFreeThenPricePerPeriod(res: Resources): String?
 
-    // Simple subscriptions do not have a trial phase
+    // Simple subscriptions do not have a offer phase
     class Simple(
         override val tier: SubscriptionTier,
         override val recurringPricingPhase: RecurringSubscriptionPricingPhase,
         override val productDetails: ProductDetails,
         override val offerToken: String,
     ) : Subscription {
-        override val trialPricingPhase = null
+        override val offerPricingPhase = null
         override fun numFreeThenPricePerPeriod(res: Resources): String? = null
         override fun tryFreeThenPricePerPeriod(res: Resources): String? = null
     }
 
-    class WithTrial(
+    class WithOffer(
```

Comment:
Would it make sense to design it this way?

```kotlin
sealed interface Subscription {
    class Simple : Subscription
    sealed class WithOffer : Subscription
    class Trial : WithOffer()
    class Intro : WithOffer()
}
```

`WithOffer` can be a `selaed class` or a `selaed interface` depending on what makes more sense - I don't know off the top of my head. I'm worried that introducing boolean states likes `isTrial()` and `isIntroOffer()` makes us lose the benefits of selaed types.

Reply:
Yeah! I agree with and looks better. See: [d017a07](https://github.com/Automattic/pocket-casts-android/pull/1722/commits/d017a07871b37bf4fe2ae7572c65a334598b50dd)